### PR TITLE
Added security protocol at top level

### PR DESCRIPTION
**Pull request by psav**

### CODE REVIEW

**Review by adamrdrew — APPROVED** [review]
Left 2 questions (1 twice, really.) If those are all good I'm all good with this, but with 2 follow on questions:

- Does this need an ADR?
- I didn't see any logic changes? Is this just because it is the broker config that cares about this, not any of the clowder code?

**Review by bsquizz — APPROVED** [review]
LGTM -- so let me sum up my understanding of what will be in cdappconfig.json -- we have one of these 3 cases:

1: default: `securityProtocol` key is present but is an empty string... (or... will the key be entirely absent? Unclear to me)
2: `securityProtocol` key is set to `"SSL"` when kafka provider is in app-interface mode and `forceTLS` is `true` in the ClowdEnvironment config.
3: `securityProtocol` key is set to `"SASL_SSL"` when kafka provider is 'managed', `managed_ephem`, or `strimzi`

File: controllers/cloud.redhat.com/providers/kafka/managed-ephem.go [hunk 1]
```diff
@@ -85,6 +85,7 @@ func (mep *managedEphemProvider) Provide(app *crd.ClowdApp) error {
 				SecurityProtocol: utils.StringPtr("SASL_SSL"),
```
[the remark this answers is not in the document]
Comment (bsquizz) [reply controllers/cloud.redhat.com/providers/kafka/managed-ephem.go:85]: My understanding is the field is only being deprecated -- so it is remaining there for now (yeah, for backward compat).

[the remark this answers is not in the document]
Comment (psav) [reply controllers/cloud.redhat.com/providers/kafka/managed-ephem.go:85]: Yup needed for deprecation

File: controllers/cloud.redhat.com/providers/kafka/managed.go [hunk 2]
```diff
@@ -105,6 +105,7 @@ func (k *managedKafkaProvider) getBrokerConfig(secret *core.Secret) (config.Brok
 		SecurityProtocol: utils.StringPtr("SASL_SSL"),
```
[the remark this answers is not in the document]
Comment (psav) [reply controllers/cloud.redhat.com/providers/kafka/managed.go:105]: Yup needed for deprecation

**Comment by psav** [comment]
> LGTM -- so let me sum up my understanding of what will be in cdappconfig.json -- we have one of these 3 cases:
> 
> 1: default: `securityProtocol` key is present but is an empty string... (or... will the key be entirely absent? Unclear to me) 2: `securityProtocol` key is set to `"SSL"` when kafka provider is in app-interface mode and `forceTLS` is `true` in the ClowdEnvironment config. 3: `securityProtocol` key is set to `"SASL_SSL"` when kafka provider is 'managed', `managed_ephem`, or `strimzi`

Correct on all accounts, I believe the key will not exist if empty

**Pull request merged**